### PR TITLE
[Fix #29] Add rules for using rspec subjects

### LINKED ISSUE #29: Do not check subject and use `is_expected` instead of it (Rspec)
The general rule looks like: do not use `expect(subject)` or `expect(subject_name` and use `is_expected` instead of it. 

```ruby
# first example - bad

subject { service.call }

it 'test' do
  expect(subject).to eq value
end  

# second example -  bad

subject(:service_call) { service.call }

it 'test' do
  expect(service_call).to eq value
end  

# third example - good

subject { service.call }

it 'test' do
  is_expected.to eq value
end  

# fourth example - also good

subject(:service_call) { service.call }

it 'test' do
  expect(service_call.valid?).to be true
end  

# fifth example - also good

subject(:service_call) { service.call }

it 'test' do
  expect { service_call }.to change { User.count }
end  
```

I can find only this cop https://rubocop-rspec.readthedocs.io/en/latest/cops_rspec/#rspecnamedsubject which restrict using `subject` in tests and ask to use the subject name or `is_expected` (so the second example is correct). Also, I couldn't find related issues with the question. 

So my question: 

- what do you think about this rule? Does the second example of sense?
- do you know some other issues or cops with this topic?

### PR DESCRIPTION
**Pull request by roman-dubrovsky**
Close #29

### CODE REVIEW

File: default.yml [hunk 1]
```diff
@@ -41,5 +41,8 @@ Metrics/BlockLength:
 Metrics/LineLength:
   Max: 120
 
+RSpec/ImplicitSubject:
+  Enabled: false
```
Comment (AleksSenkou) [inline default.yml]: Is it possible to configure it instead of disabling?

Comment (roman-dubrovsky) [reply default.yml]: @AleksSenkou what do you mean about configuring? + all decisions have been discussed in the  issue

Comment (AleksSenkou) [reply default.yml]: @roman-dubrovsky my point here is that we just disable a rule, but not configure it.

Will this one:
```ruby
subject { service.call }

it 'test' do
  expect(subject).to eq value
end  
```
raise an error?

Comment (roman-dubrovsky) [reply default.yml]: and, it will but because `subject` does not have a name

Comment (roman-dubrovsky) [reply default.yml]: this is  a correct example 

```
subject(:tested_value) { service.call }

it 'test' do
  expect(tested_value).to eq value
end  
```

Comment (AleksSenkou) [reply default.yml]: Got it 👍

**Pull request merged**